modules/renderer/background.js

```diff
@@ -187,7 +188,7 @@ export function Background(context) {
         }
 
         var q = stringQs(location.hash.substring(1)),
-            chosen = q.background || q.layer,
+            chosen = q.background || q.layer || context.storage('background'),
```

Comment:
The problem with putting it here is that now the user's preferred background might not be visible where they started editing the map.  This is especially a problem if a user is entering iD via a task manager or QA tool, and editing in a place that is not where they normally use iD.

Could you move the `context.storage` code down to the block where `best` gets set... Then in there set a variable called `recent` that contains this BackgroundSource only if it is valid in the current visible extent.

Then a few lines later do `background.baseLayerSource(findSource(chosen) || recent || best || ...etc)`